gomock/call.go

```diff
@@ -245,15 +245,26 @@ func (c *Call) matches(args []interface{}) error {
 				// sample: Foo(a int, b int, c ...int)
 
 				if len(c.args) == len(args) {
-					if c.args[i].Matches(args[i]) {
+					if m.Matches(args[i]) {
 						// Got Foo(a, b, c) want Foo(matcherA, matcherB, gomock.Any())
 						// Got Foo(a, b, c) want Foo(matcherA, matcherB, someSliceMatcher)
 						// Got Foo(a, b, c) want Foo(matcherA, matcherB, matcherC)
 						// Got Foo(a, b) want Foo(matcherA, matcherB)
 						// Got Foo(a, b, c, d) want Foo(matcherA, matcherB, matcherC, matcherD)
 						break
 					}
-				} else if c.args[i].Matches(args[i:]) {
+				}
+				// The number of actual args don't match the number of matchers.
+				// If this function still matches it is because the last matcher
+				// matches all the remaining arguments or the lack of any.
+				// Convert the remaining arguments, if any, into a slice of the
+				// expected type.
+				vargsType := c.methodType.In(c.methodType.NumIn() - 1)
```

Comment:
I think this should be c.methodType.In(c.methodType.NumIn() - 1).Elem() otherwise 263 is creating a slice of slices.

Reply:
It turns out that MakeSlice wants a slice type.